api4/post.go

```diff
@@ -384,8 +384,11 @@ func getPostThread(c *Context, w http.ResponseWriter, r *http.Request) {
 	if c.Err != nil {
 		return
 	}
-
-	list, err := c.App.GetPostThread(c.Params.PostId)
+	skipFetchThreads := false
+	if r.URL.Query().Get("fetchThreads") == "false" {
+		skipFetchThreads = true
```

Comment:
Why should `skipFetchThreads` ever be allowed to be `true` for the API named `getPostThread`?

Reply:
@lieut-data - I know it doesn't make alot of sense due to the naming, but downstream getPostThread is fetching posts and roots. I propagate `skipFetchThreads` downstream to properly fetch and set ReplyCount on the root posts.

Reply:
Hmm, I confess I'm still confused: if `skipFetchThreads` is `true`, we don't fetch the content of the thread: just the root + reply count + possible the post in question. Is that ever the desired behaviour of the caller? I presumed this API endpoint would be used to fetch the full contents of the thread: when would it ever be called otherwise?

I could understand hard-coding the value of `skipFetchThreads` to be `false` here, but I'm not sure why we're accepting it from the client in this case.

Reply:
@lieut-data - passing of the parameter from outside was done to maintain compatibility with older clients. If we hardcode `skipFetchThreads`, I am not sure of the exact scenario - but the Mobile apps might perform this API call and expect the returned value to contain more than just a post and it's root.
Take a look at: https://github.com/mattermost/mattermost-server/blob/b29b9331d716191cdcf112433ad20070724ed3ec/store/sqlstore/post_store.go#L307
This is the original behavior and it selects ALL posts with the same RootId.
So the behavior is quite different.

Reply:
I'm proposing that we hardcode `skipFetchThreads = false`, since this API doesn't ever make sense to call with `skipFetchThreads = true`.

Reply:
(Both for old clients, who don't know about the parameter, and new clients, who wouldn't ask for threads and not ask for threads at the same time.)

Reply:
@lieut-data - are you positive this won't break anything in mattermost-mobile apps? I am less familiar with their API usage...

Reply:
@reflog, this is a new parameter, right? So hard-coding it to it's current behaviour can't "break" anything unless we're also making changes to the clients to expect otherwise. Or am I missing something here?

Reply:
@reflog and I synched off offline -- this app method is poorly named. I'd love to revisit the post store overall: it's very confusing :D